build/Publish.targets

```diff
@@ -112,16 +112,23 @@
         <Overwrite>true</Overwrite>
       </FilesToPublish>
 
-      <PackagesToPublish Include="$(DependencyPackagesDir)*.nupkg" Exclude="$(DependencyPackagesDir)*.symbols.nupkg" />
-      <PackagesToPublish Include="$(DependencyPackagesDir)*.symbols.nupkg" Exclude="@(PackagesToPublish)" IsSymbolsPackage="true" />
+      <_SymbolArtifactPath Include=">$(DependencyPackagesDir)%(PackageArtifact.Identity).*$(BuildNumber).nupkg" Artifact="%(PackageArtifact.Identity)" />
+      <_PackageArtifactPath Include=">$(DependencyPackagesDir)%(PackageArtifact.Identity).*$(BuildNumber).nupkg" Exclude="@(_SymbolArtifactPath)" Artifact="%(PackageArtifact.Identity)" />
```

Comment:
@natemcmaster this would be a fair assumption to make, no?

Reply:
For most builds it would be okay, but when IsFinalBuild=true, BuildNumber won't be in the file name.

To split these apart, we probably need to use the `ArtifactInfo` item group, which includes info about package ID + version. You could then use that to compute all the file paths you need `$(DependencyPackagesDir)%(ArtifactInfo.PackageId).$(ArtifactInfo.Version).nupkg`. Unfortunately, that doesn't contain package category due to #686, so you would need to join PackageArtifact and ArtifactInfo. That's hard to do in MSBuild, so I have a task for such a thing:
https://github.com/aspnet/Universe/blob/namc/graph-build/build/tasks/JoinItems.cs
Example usage:
https://github.com/aspnet/Universe/blob/6029a73327d3eb886b3c2d3957a06b8d2450c069/build/targets/Repository.Common.targets#L166-L170

I'm about to add this task to release/2.1 because I need it to solve #803

Reply:
Gotcha. Would you like to pick this up once you've made the changes? Would be easier if one of one partied around with BuildTools + this repo changes?

Reply:
If you're okay waiting for me to fix #803 (almost ready to open a PR with a fix), lets do that. Otherwise, feel free to just copy https://github.com/aspnet/Universe/blob/namc/graph-build/build/tasks/JoinItems.cs to your PR. I'm going to add it verbatim to my PR to fix 803

Reply:
Sure, I think we can wait a bit. @dagood we should be ready with this change by tomorrow morning.

Reply:
Sounds good, thanks for the update!